### ISSUE TITLE
secret-manager client 2.18 incompatible with google-auth 2.24

### ISSUE THREAD

**Issue by kojiromike** [googleapis/google-cloud-python#12364]
When we had google-cloud-secret-manager 2.18 and google-auth 2.24, we got this traceback:

```
  File "/opt/venv/lib/python3.11/site-packages/phc/secrets/lib.py", line 18, in get
    response = self.client.access_secret_version(request={"name": name})
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/venv/lib/python3.11/site-packages/google/cloud/secretmanager_v1/services/secret_manager_service/client.py", line 1764, in access_secret_version
    self._validate_universe_domain()
  File "/opt/venv/lib/python3.11/site-packages/google/cloud/secretmanager_v1/services/secret_manager_service/client.py", line 551, in _validate_universe_domain
    or SecretManagerServiceClient._compare_universes(
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/venv/lib/python3.11/site-packages/google/cloud/secretmanager_v1/services/secret_manager_service/client.py", line 528, in _compare_universes
    credentials_universe = getattr(credentials, "universe_domain", default_universe)
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/venv/lib/python3.11/site-packages/google/auth/compute_engine/credentials.py", line 139, in universe_domain
    self._universe_domain = _metadata.get_universe_domain()
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: get_universe_domain() missing 1 required positional argument: 'request'
```

This was resolved by updating google-auth to 2.28.1. I didn't test intermediate versions.

I would expect pip to complain about installing google-cloud-secret-manager 2.18 with an incompatible version of google-auth.

**Comment by parthea** [comment]
Hi @kojiromike,

I believe the issue you're seeing was introduced in `google-auth==2.24.0` and fixed in version 2.25.1 and newer. 

https://github.com/googleapis/google-auth-library-python/releases/tag/v2.24.0
https://github.com/googleapis/google-auth-library-python/releases/tag/v2.25.1

Please can you check if the issue still appears with `google-auth` `2.25.1` or newer?

**Comment by parthea** [comment]
> This was resolved by updating google-auth to 2.28.1

`google-auth>=2.14.1` will work (excluding `2.24.0` and `2.25.0` due to a bug in `google-auth`)
